packages/components-library/src/components/filters/ChangeFilters.vue

```diff
@@ -0,0 +1,200 @@
+library.add(faCaretRight, faTimes)
+
+export default Vue.extend({
+  name: 'ChangeFilters',
+  components: { FontAwesomeIcon },
+  props: {
+    /**
+     * The filters to select from
+     * @model
+     */
+    filters: {
+      type: Array,
+      required: true
+    },
+    /**
+     * The selected filters
+     * @model
+     */
+    value: {
+      type: Array,
+      required: true
+    }
+  },
+  data () {
+    return {
+      selected: this.value
+    }
+  },
+  watch: {
+    value: {
+      handler (newValue) {
+        this.selected = [...newValue]
+      },
+      immediate: true
+    }
+  },
+  methods: {
+    unique: (value, index, items) => items.indexOf(value) === index,
+    checkboxHandler (option, state) {
+      if (state) {
+        this.selected.push(option.name)
+      } else {
+        this.selected = this.selected.filter((item) => item !== option.name)
+      }
+
+      // parent compound toggle
+      if (option.type === 'compound-title') {
+        const subCompounds = this.filters
+          .filter((item) => item.compound === option.name)
+          .map((item) => item.name)
+        if (this.selected.includes(option.name)) {
+          // Select all and make unique
+          this.selected = [...this.selected, ...subCompounds].filter(
+            this.unique
+          )
+        } else {
+          // Deselect all
+          this.selected = this.selected.filter(
+            (item) => !subCompounds.includes(item)
+          )
+        }
+      }
+
+      // child compound toggle
+      if (option.compound) {
+        const subCompounds = this.filters
+          .filter((item) => item.compound === option.compound)
+          .map((item) => item.name)
+        const allSelected = subCompounds.every((item) =>
+          this.selected.includes(item)
+        )
+        const noneSelected = subCompounds.every(
+          (item) => !this.selected.includes(item)
+        )
+        const partialSelected = !allSelected && !noneSelected
+        if (allSelected) {
+          this.selected = [option.compound, ...this.selected].filter(
+            this.unique
+          )
+        }
+        if (noneSelected) {
+          this.selected = this.selected.filter(
+            (item) => item !== option.compound
+          )
+        }
+        if (partialSelected) {
+          this.selected = this.selected.filter(
+            (item) => item !== option.compound
+          )
+        }
+      }
+      this.$emit('input', this.selected)
+    }
+  }
+})
+</script>
+
+<style scoped>
+.remove-button {
+  transition: color 0.2s;
+  height: inherit;
+  width: 1.5em;
+  text-align: center;
+  display: inline-block;
+  position: absolute;
+  right: 10px;
+  cursor: pointer;
+}
+.remove-button:hover {
+  color: var(--danger);
+}
+</style>
+
+<docs>
+Filter selector
+### Usage
+```jsx
+const filters = [
+  {
+    name: 'color',
+    label: 'Color',
+  },
+  {
+    name: 'Region',
+    label: 'region',
```

Comment:
Label should be upper-case instead of name?

Reply:
updated case